src/coreclr/runtime/arm64/WriteBarriers.S

```diff
@@ -1,7 +1,15 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-#include <unixasmmacros.inc>
+#include "AsmMacros_Shared.h"
+
+#if defined(__APPLE__)
+    // Currently the build is failing without this due to an issue if the first method in the assembly file has an alternate entry at the start of the file.
+    // Fix, but adding an empty, unused method
+    LEAF_ENTRY RhpWriteBarriersDoNotFailToBuild, _TEXT
+       ret
+    LEAF_END RhpWriteBarriersDoNotFailToBuild, _TEXT
+#endif
```

Comment:
For posterity, this was reported as FB14743667 to Apple. It should have be fixed in Xcode 16.3. We should still keep the workaround until we bump the minimum Xcode requirement past this version.

The general issue is that the new linker (ld-prime) didn't correctly handle combination of symbols at overlapping addresses and alt-entry symbols. There are some cases where section symbols are implicitly generated at the beginning of each section in object file so it's easy to run into this. Old linker (ld64) doesn’t handle alt-entry symbols at a start of a section so it's problematic there too.